src/main/scala/scorex/core/network/peer/PeerManager.scala

```diff
@@ -152,7 +155,8 @@ object PeerManager {
                           blacklistedPeers: Seq[InetAddress],
                           sc: ScorexContext): Option[PeerInfo] = {
         val candidates = knownPeers.values.filterNot { p =>
-          excludedPeers.exists(_.peerSpec.address == p.peerSpec.address)
+          excludedPeers.exists(_.peerSpec.address == p.peerSpec.address) &&
+            blacklistedPeers.exists(addr => p.peerSpec.address.map(_.getAddress).contains(addr))
```

Comment:
Mind blowing naming, when trying to make sense out of it `... address.map(_.getAddress) ...`

Reply:
What do you propose?

Reply:
Renaming, but this is not related to this PR